cpp/include/raft/linalg/detail/eig.cuh

```diff
@@ -91,6 +91,15 @@ void eigDC(raft::resources const& handle,
 #if CUDART_VERSION < 11010
   eigDC_legacy(handle, in, n_rows, n_cols, eig_vectors, eig_vals, stream);
 #else
+
+  // Use a new stream instead of `cudaStreamPerThread` to avoid cusolver bug # 4580093.
```

Comment:
IIUC, the cusolver bug is solved in cuda toolkit 12.4.1.003. It would be great if we apply this workaround only when we use an earlier cuda version.